include/sdf/ParticleEmitter.hh

```diff
@@ -0,0 +1,352 @@
+    /// color values begins from the left side of the image and moves to the
+    /// right over the lifetime of the particle, therefore only the horizontal
+    /// dimension of the image is used.
+    /// The ColorRangeImage has higher priority than ColorEnd and
+    /// ColorStart. If all three are set, ColorRangeImage should be used.
+    public: std::string ColorRangeImage() const;
+
+    /// \brief Set the path to the color image used as an affector.
+    /// \param[in] _image The path to the color image.
+    /// \sa std::String ColorRangeImage()
+    public: void SetColorRangeImage(const std::string &_image);
+
+    /// \brief Get the topic used to update the particle emitter properties.
+    /// \return The topic used to update the particle emitter.
+    public: std::string Topic() const;
+
+    /// \brief Set the topic used to update the particle emitter properties.
+    /// \param[in] _topic The topic used to update the particle emitter.
+    public: void SetTopic(const std::string &_topic);
+
+    /// \brief Get the pose of the particle emitter. This is the pose of the
+    /// emitter as specified in SDF
+    /// (<particle_emitter><pose> ... </pose></particle_emitter>).
+    /// \return The pose of the particle emitter.
+    public: const ignition::math::Pose3d &RawPose() const;
+
+    /// \brief Set the pose of the particle emitter object.
+    /// \sa const ignition::math::Pose3d &RawPose() const
+    /// \param[in] _pose The pose of the particle emitter.
+    public: void SetRawPose(const ignition::math::Pose3d &_pose);
+
+    /// \brief Get the name of the coordinate frame relative to which this
+    /// emitter's pose is expressed. An empty value indicates that the frame is
+    /// relative to the parent link.
+    /// \return The name of the pose relative-to frame.
+    public: const std::string &PoseRelativeTo() const;
+
+    /// \brief Set the name of the coordinate frame relative to which this
+    /// emitter's pose is expressed. An empty value indicates that the frame is
+    /// relative to the parent link.
+    /// \param[in] _frame The name of the pose relative-to frame.
+    public: void SetPoseRelativeTo(const std::string &_frame);
+
+    /// \brief Get SemanticPose object of this object to aid in resolving poses.
+    /// \return SemanticPose object for this emitter.
+    public: sdf::SemanticPose SemanticPose() const;
+
+    /// \brief Get a pointer to the SDF element that was used during load.
+    /// \return SDF element pointer. The value will be nullptr if Load has
+    /// not been called.
+    public: sdf::ElementPtr Element() const;
+
+    /// \brief Get a pointer to the emitter's material properties. This can
+    /// be a nullptr if material properties have not been set.
+    /// \return Pointer to the emitter's material properties. Nullptr
+    /// indicates that material properties have not been set.
+    public: sdf::Material *Material() const;
```

Comment:
Should this return `const sdf::Material *`?

Reply:
Updated in 300e52f655d536661797edfe801dbb41cb67b3b8